router/batchrouter/batchrouter.go

```diff
@@ -759,19 +768,23 @@ func (brt *HandleT) copyJobsToStorage(provider string, batchJobs *BatchJobsT, ma
 	}
 }
 
-func (brt *HandleT) sendJobsToStorage(provider string, batchJobs BatchJobsT, config map[string]interface{}, makeJournalEntry bool, isAsync bool) {
+func (brt *HandleT) sendJobsToStorage(batchJobs BatchJobsT) {
 	if disableEgress {
 		out := asyncdestinationmanager.AsyncUploadOutput{}
 		for _, job := range batchJobs.Jobs {
 			out.SucceededJobIDs = append(out.SucceededJobIDs, job.JobID)
 			out.SuccessResponse = fmt.Sprintf(`{"error":"%s"`, rterror.DisabledEgress.Error())
 		}
-		brt.setMultipleJobStatus(out)
+
+		// rsources stats
+		rsourcesStats := rsources.NewStatsCollector(brt.rsourcesService)
+		rsourcesStats.BeginProcessing(batchJobs.Jobs)
+
+		brt.setMultipleJobStatus(out, rsourcesStats)
 		return
 	}
 
 	destinationID := batchJobs.BatchDestination.Destination.ID
```

Comment:
note: It seems we have a lot of `destinationID` specific logic, we should invest in creating a new struct that handles this specific logic.

This struct could leave inside `brt.asyncDestinationStruct`, it will eliminate a lot of `brt.asyncDestinationStruct[destinationID]` look ups, as well as making the code safer and more readable